packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -1915,168 +2178,491 @@ class TimePickerDialog extends StatefulWidget {
+      ),
+      child: Padding(
+        padding: pickerTheme.padding ?? defaultTheme.padding,
+        child: LayoutBuilder(builder: (BuildContext context, BoxConstraints constraints) {
+          final Size constrainedSize = constraints.constrain(dialogSize);
+          final Size allowedSize = Size(
+            constrainedSize.width < minDialogSize.width ? minDialogSize.width : constrainedSize.width,
+            constrainedSize.height < minDialogSize.height ? minDialogSize.height : constrainedSize.height,
+          );
+          return SingleChildScrollView(
+            scrollDirection: Axis.horizontal,
+            child: SingleChildScrollView(
+              child: AnimatedContainer(
+                width: allowedSize.width,
+                height: allowedSize.height,
+                duration: _kDialogSizeAnimationDuration,
+                curve: Curves.easeIn,
+                child: Column(
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: <Widget>[
+                    Padding(
+                      padding: EdgeInsetsDirectional.only(start: theme.useMaterial3 ? 0 : 16, bottom: theme.useMaterial3 ? 20 : 24),
+                      child: Text(
+                        widget.helpText ?? helpText,
+                        style: pickerTheme.helpTextStyle ?? defaultTheme.helpTextStyle,
+                      ),
+                    ),
+                    Expanded(
+                      child: Form(
+                        key: _formKey,
+                        autovalidateMode: _autovalidateMode.value,
+                        child: Padding(
+                          padding: pickerPadding,
+                          child: _TimePicker(
+                            time: widget.initialTime,
+                            onTimeChanged: _handleTimeChanged,
+                            cancelText: widget.cancelText,
+                            confirmText: widget.confirmText,
+                            errorInvalidText: widget.errorInvalidText,
+                            hourLabelText: widget.hourLabelText,
+                            minuteLabelText: widget.minuteLabelText,
+                            restorationId: 'time_picker',
+                            entryMode: _entryMode.value,
+                            orientation: widget.orientation,
+                            onEntryModeChanged: _handleEntryModeChanged,
+                          ),
+                        ),
+                      ),
+                    ),
+                    actions,
+                  ],
+                ),
+              ),
+            ),
+          );
+        }),
+      ),
+    );
+  }
+}
+
+// The _TimePicker widget is constructed so that in the future we could expose
+// this as a public API for embedding time pickers into other non-dialog
+// widgets, once we're sure we want to support that.
```

Comment:
Nice. We have in the past seen similar requests for exposing parts of the date picker as well.

Reply:
Yeah, I figured I would arrange it this way, and if/when we decide to make it available seprately, we could do the date picker at the same time.